src/book-test-instance.py

```diff
@@ -0,0 +1,393 @@
+#!/usr/bin/env python3
+
+import os
+import json
+import requests
+from requests.auth import HTTPBasicAuth
+from requests_oauthlib import OAuth1
+import sys
+import re
+import argparse
+import tempfile
+import hashlib
+
+
+JIRA_API = 'https://jira.greenpeace.org/rest/api/2'
+GITHUB_API = 'https://api.github.com'
+SWARM_API = 'https://us-central1-planet-4-151612.cloudfunctions.net/p4-test-swarm'
+
+INSTANCE_REPO_PREFIX = 'greenpeace/planet4-test-'
+JIRA_INSTANCE_FIELD = 'customfield_13000'
+
+
+def get_pull_request(pr_number=None, repository=None, pr_url=None):
+    """
+    Fetch PR details
+
+    Extract data from PR url
+    Request PR data
+    """
+
+    if pr_url:
+        regex = re.compile('https://github.com/(.*)/pull/([0-9]{1,6})')
+        matches = regex.match(pr_url)
+        vprint('Parsing URL {0}'.format(matches.groups()))
+
+        repository = matches[1] if not repository else repository
+        pr_number = matches[2] if not pr_number else pr_number
+
+    if not repository or not pr_number:
+        raise Exception('PR id could not be parsed.')
+
+    pr_endpoint = '{0}/repos/{1}/pulls/{2}'.format(
```

Comment:
Oh no, I'm doing this in python3, python2 isn't maintained anymore, can we update the builder ?

Reply:
Perhaps, it has python because that is already in the base image it uses https://hub.docker.com/layers/cimg/php/7.3-node/images/sha256-b9d6c0c9b5de3b8fa023d9ca122c74c1899eb18d4535a3c15b68d69a3b748655?context=explore

Reply:
The layer that adds python is this one
https://github.com/CircleCI-Public/circleci-dockerfiles/blob/master/php/images/7.3.23-fpm-stretch/Dockerfile#L27-L35

Reply:
We could add it, but that wil add a layer to the image (or increase the size of an existing) and make it even slower to spin up. I guess if it works with 2 we can use that for now?

Reply:
I just ran this one https://hub.docker.com/layers/cimg/php/7.3.26-node/images/sha256-b9d6c0c9b5de3b8fa023d9ca122c74c1899eb18d4535a3c15b68d69a3b748655?context=explore which has python3 (as `python3`), I don't know if we're using the latest circleci images :thinking:  but I can adapt the script to python2, working on it :ok_hand:

Reply:
We do start from the latest circleci... kinda :smile: The builder starts `FROM greenpeaceinternational/circleci-base:latest`, which in turns starts `FROM circleci/php:7.3-node`. So the base image is only updated when `planet4-circleci` is updated, which is already a while ago. I made a PR to trigger a new build (pinned the version in the meanwhile): https://github.com/greenpeace/planet4-circleci/pull/40

Reply:
Turns out we needed to change the base image, the docker hub repo we looked at above was actually another CircleCI repo than the one we were using. So I made [a new PR to switch to that as base image](https://github.com/greenpeace/planet4-circleci/pull/41), which seems to pass. But let's try using python2 first, since there's a looot of things to check if they would work with the new image.

Once we confirmed the new image we can change this script to make use of python3 goodies.